pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -232,23 +232,25 @@ func (s *Server) initConfigSources(args *PilotArgs) (err error) {
 			s.ConfigStores = append(s.ConfigStores, configController)
 			log.Infof("Started File configSource %s", configSource.Address)
 		case XDS:
-			xdsMCP, err := adsc.New(srcAddress.Host, &adsc.Config{
-				Namespace: args.Namespace,
-				Workload:  args.PodName,
-				Revision:  args.Revision,
-				Meta: model.NodeMetadata{
-					Generator: "api",
-					// To reduce transported data if upstream server supports. Especially for custom servers.
-					IstioRevision: args.Revision,
-				}.ToStruct(),
+			xdsMCP, err := adsc.New(srcAddress.Host, &adsc.ADSConfig{
```

Comment:
It looks like you need to be able to switch to `adsc.NewDelta` here.

Reply:
Currently it has not been used in tests or code yet, but it will be.

Reply:
Thanks, got it.